stopped dealing in absolutes

Kapparina opened the pull request:
Fixes #1

Updated the project with a 50/50 chance to have the correct answer selected; there are now 3 correct answers and 3 incorrect answers.

Kapparina committed:
Add `rand` crate as a dependency

The `rand` crate is now included in `Cargo.toml` to provide random number generation functionality. This will enable us to utilize its utilities across our project where randomness is required.

Kapparina committed:
Enhance answer validation logic by randomizing responses

Introduced randomized answer selection for the given question using `rand` crate, replacing static answer validation. Updated response messages to handle new possible outcomes.

Kapparina commented:
I showed you my PR, please answer me.

Kapparina committed:
Add system time check before main logic

Introduced functionality to check the current system time and perform actions based on whether the current seconds since epoch are divisible by 5. The code now prints distinct messages depending on the result of the time check.

Kapparina committed:
Change question text to include "and everything in it".

Updated the question to be more precise. This modification ensures consistency between the prompt and the expected input in the `the_answer` function.

Signed-off-by: Kapparina <116474667+Kapparina@users.noreply.github.com>

MrDwarf7 commented on src/main.rs:
```diff
@@ -1,20 +1,49 @@
+        Ok(duration) => {
+            let seconds_since = duration.as_secs();
+            if seconds_since % 5 == 0 {
+                println!("UWUUUUUUUUUUUUUUUUUUUUU");
```
Need a better message for when the modulo isn't valid

Kapparina replied:
Surely you're a fan of Doctor Strange?

MrDwarf7 commented on src/main.rs:
```diff
@@ -1,20 +1,49 @@
+                println!("YOU HAVE COMMITTED HERESY!");
+            }
+        }
+        Err(_) => {
```
Implement a custom Error enum that impls Error from the std to handle this, and remove the direct error call in favor of a question mark

Kapparina replied:
Job done

MrDwarf7 commented on src/main.rs:
```diff
@@ -1,20 +1,49 @@
 }
+
+fn draw_answer_from_hat() -> i32 {
+    let numbers = [1, 2, 3, 42, 69, 9001];
```
Array is missing 420

MrDwarf7 commented on src/main.rs:
```diff
@@ -1,20 +1,49 @@
+
+fn draw_answer_from_hat() -> i32 {
+    let numbers = [1, 2, 3, 42, 69, 9001];
+    let mut rng = thread_rng();
```
Please use xoshiro rng generators for this project

Kapparina replied:
That sounds like an absolute, only a Sith deals in absolutes.

MrDwarf7 reviewed (changes requested):
Once issues have been resolved this can be merged into main.
If any assistance please follow up

Kapparina committed:
Add custom error handling and update main return type

Introduced `ButterError` for custom error messages and updated `main` to return `Result`. Implemented the `From` trait for `SystemTimeError` and added informative error messages to improve debugging.

Signed-off-by: Kapparina <116474667+Kapparina@users.noreply.github.com>

MrDwarf7 commented:
lgtm
